### DISCUSSION TITLE
Creating an IssuedDocument with the C# SDK

### DISCUSSION THREAD

**Discussion by ghost** [fattureincloud/api#312] [Q&A] [answered]
The constructor of the IssuedDocument class takes an Entity as the second parameter.
For an invoice that is the client the invoice is for.
When I retrieve a list of existing Clients through the ClientApi I receive a List of ModelClient objects.

Because I have to send the correct Client data with the invoice, I have to transform the ModelClient into an Entity.

Am I doing the right thing here?
If so, is there a better way to do the transformation?

**Comment by valmoz** [comment, marked as answer]
Hi @EmondW 
yes, in our model, clients and suppliers are both entities, but even if most of the arguments are shared, they also have some different ones.
That's why we preferred to keep different APIs for both of them, but in the case of the Invoice creation the generator had some problems managing the two kinds of entities, so we still preferred to create an entity model to be used in that case.

You just need to copy all the arguments of your interest from the client into the entity and ignore the additional ones, and you should be fine.

Thanks
Mauro

**Reply by ghost** [reply]
Thank you!

Yes, I suspected some kind of inheritance hierarchy but that's not easy for (de)serializers.
